phys2bids/phys2bids.py

```diff
@@ -259,13 +259,12 @@ def phys2bids(filename, info=False, indir='.', outdir='.', heur_file=None,
     if info:
         return
 
-    # Run analysis on trigger channel to get first timepoint and the time offset.
-    # #!# Get option of no trigger! (which is wrong practice or Respiract)
-    phys_in.check_trigger_amount(chtrig, thr, num_timepoints_expected, tr)
-
     # Create trigger plot. If possible, to have multiple outputs in the same
     # place, adds sub and ses label.
     if tr != 0 or num_timepoints_expected != 0:
+        # Run analysis on trigger channel to get first timepoint and the time offset.
+        # #!# Get option of no trigger! (which is wrong practice or Respiract)
```

Comment:
I just detected a bug, if ``-tr`` is not inputed and ``-ntp`` is inputed this happens:
```
phys2bids -in tutorial_file.txt -chtrig 1 -outdir outdir -ntp 158 
INFO:phys2bids.phys2bids:Currently running phys2bids version v1.3.0-beta+335.gca4394e.dirty
INFO:phys2bids.phys2bids:Input file is tutorial_file.txt
INFO:phys2bids.utils:File extension is .txt
WARNING:phys2bids.utils:If both acq and txt files exist in the path, acq will be selected.
INFO:phys2bids.phys2bids:Reading the file ./tutorial_file.txt
INFO:phys2bids.interfaces.txt:phys2bids detected that your file is in Labchart format
INFO:phys2bids.phys2bids:Reading infos
INFO:phys2bids.physio_obj:
------------------------------------------------
File tutorial_file.txt contains:
01. Trigger; sampled at 1000.0 Hz
02. CO2; sampled at 1000.0 Hz
03. O2; sampled at 1000.0 Hz
04. Pulse; sampled at 1000.0 Hz
------------------------------------------------

INFO:phys2bids.physio_obj:Counting trigger points
INFO:phys2bids.physio_obj:The number of timepoints according to the std method is 158 and the threshold is 1.1523587407910223
INFO:phys2bids.physio_obj:Checking number of timepoints
INFO:phys2bids.physio_obj:Found just the right amount of timepoints!
INFO:phys2bids.phys2bids:Plot trigger
/home/vicente/phys2bids/phys2bids/viz.py:72: UserWarning: Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.
  subplot.set_xlim([-tr * 4, tr * 4])
/home/vicente/phys2bids/phys2bids/viz.py:48: RuntimeWarning: invalid value encountered in true_divide
  return x / tr
/home/vicente/phys2bids/phys2bids/viz.py:48: RuntimeWarning: invalid value encountered in true_divide
  return x / tr
/home/vicente/phys2bids/phys2bids/viz.py:48: RuntimeWarning: divide by zero encountered in true_divide
  return x / tr
Traceback (most recent call last):
  File "/home/vicente/.local/bin/phys2bids", line 11, in <module>
    load_entry_point('phys2bids', 'console_scripts', 'phys2bids')()
  File "/home/vicente/phys2bids/phys2bids/phys2bids.py", line 362, in _main
    phys2bids(**vars(options))
  File "/home/vicente/phys2bids/phys2bids/phys2bids.py", line 276, in phys2bids
    plot_path, tr, phys_in.thr, num_timepoints_expected, filename)
  File "/home/vicente/phys2bids/phys2bids/viz.py", line 113, in plot_trigger
    plt.savefig(fileprefix + '_trigger_time.png', dpi=dpi)
  File "/home/vicente/.local/lib/python3.6/site-packages/matplotlib/pyplot.py", line 729, in savefig
    res = fig.savefig(*args, **kwargs)
  File "/home/vicente/.local/lib/python3.6/site-packages/matplotlib/figure.py", line 2180, in savefig
    self.canvas.print_figure(fname, **kwargs)
  File "/home/vicente/.local/lib/python3.6/site-packages/matplotlib/backend_bases.py", line 2091, in print_figure
    **kwargs)
  File "/home/vicente/.local/lib/python3.6/site-packages/matplotlib/backends/backend_gtk3agg.py", line 82, in print_png
    return agg.print_png(filename, *args, **kwargs)
  File "/home/vicente/.local/lib/python3.6/site-packages/matplotlib/backends/backend_agg.py", line 527, in print_png
    FigureCanvasAgg.draw(self)
  File "/home/vicente/.local/lib/python3.6/site-packages/matplotlib/backends/backend_agg.py", line 388, in draw
    self.figure.draw(self.renderer)
  File "/home/vicente/.local/lib/python3.6/site-packages/matplotlib/artist.py", line 38, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/home/vicente/.local/lib/python3.6/site-packages/matplotlib/figure.py", line 1691, in draw
    child.apply_aspect(pos)
  File "/home/vicente/.local/lib/python3.6/site-packages/matplotlib/axes/_secondary_axes.py", line 180, in apply_aspect
    self._set_lims()
  File "/home/vicente/.local/lib/python3.6/site-packages/matplotlib/axes/_secondary_axes.py", line 311, in _set_lims
    set_lim(lims)
  File "/home/vicente/.local/lib/python3.6/site-packages/matplotlib/axes/_base.py", line 3241, in set_xlim
    left = self._validate_converted_limits(left, self.convert_xunits)
  File "/home/vicente/.local/lib/python3.6/site-packages/matplotlib/axes/_base.py", line 3158, in _validate_converted_limits
    raise ValueError("Axis limits cannot be NaN or Inf")
ValueError: Axis limits cannot be NaN or Inf
```
I believe the best way to avoid this behavior is that instead of a or condition, we have an and condition. We could also add an if so the viz.plot_trigger function does not get ``tr=None``

Reply:
This is an error inherit from #153 we can deal with it in a bug PR

Reply:
@vinferrer you're totally right. Can you implement the `and` statement instead of the `or` in this PR?